scripts/utils/js_helpers.js

```diff
@@ -18,7 +18,57 @@ const sleep = function (milliseconds) {
   return new Promise((r) => setTimeout(r, milliseconds))
 }
 
+/**
+ * Given an array of promises, returns a promise resolving to the
+ * output of the first resolving promise of the array. If any promise
+ * has been rejected in the meantime, this failure is ignored. If all
+ * promises are rejected, returns a rejecting promise returning an
+ * error containing all error messages concatenated with "&&".
+ * If entries of the input array are not promises, they will be treated
+ * as promises resolving to the input value.
+ *
+ * This function is meant to be replaced in the future by Promise.any,
+ * which is not yet availeble on Node.js.
+ * https://tc39.es/proposal-promise-any/#sec-promise.any
+ *
+ * Unlike Promise.any, if all input promises reject then the return
+ * value is a rejected promise returning an error with a message that
+ * concatenates all promise error messages (or output values, if
+ * message is not available) in the order they appear in the input
+ * array. (Promise.any returns an AggregateError.)
+ *
+ * @param {Promise[]} promiseArray array of promises (or values)
+ * @returns {Promise} promise returning the value returned by the
+ * input promise that is the quickest to resolve
+ */
+const returnFirstNotErroring = (promiseArray) =>
+  new Promise((resolve, reject) => {
+    let resolved = false
+    const errorPromises = []
+    for (const promise of promiseArray) {
+      errorPromises.push(
+        Promise.resolve(promise).then(
+          (result) => {
+            resolved = true
+            resolve(result)
```

Comment:
Does this need to be in a `if (!resolved) { resolve(result) }`? I'm not sure how promises behave when they are resolved multiple times.

Reply:
TIL, you can `resolve` and `reject` as much as you want, only the first call matters. Maybe the flag isn't needed at all?